app/controllers/grade_entry_forms_controller.rb

```diff
@@ -381,14 +381,14 @@ def csv_upload
         end
         if totals.empty?
           totals = row
-          # Create/update the grade entry items
-          GradeEntryItem.create_or_update_from_csv_rows(
-            names,
-            totals,
-            @grade_entry_form,
-            overwrite)
           next
         end
+        # Create/update the grade entry items
+        GradeEntryItem.create_or_update_from_csv_rows(
```

Comment:
You should revert this change. This part should only happen once (when processing the first row).